core/src/eval/operation.rs

```diff
@@ -2580,6 +2580,148 @@ impl<R: ImportResolver, C: Cache> VirtualMachine<R, C> {
+                            (
+                                LocIdent::from("left_only"),
+                                Field::from(RichTerm::from(left_only)),
+                            ),
+                            (
+                                LocIdent::from("left_center"),
+                                Field::from(RichTerm::from(left_center)),
+                            ),
+                            (
+                                LocIdent::from("right_center"),
+                                Field::from(RichTerm::from(right_center)),
+                            ),
+                            (
+                                LocIdent::from("right_only"),
+                                Field::from(RichTerm::from(right_only)),
+                            ),
+                        ]),
+                        attrs: RecordAttrs::default().closurized(),
+                        sealed_tail: None,
+                    }),
+                    pos_op_inh,
+                )))
+            }
+            BinaryOp::RecordDisjointMerge => {
+                let t1 = t1.into_owned();
+                let t2 = t2.into_owned();
+
+                let Term::Record(mut record1) = t1 else {
+                    return Err(mk_type_error!(
+                        "record/disjoint_merge",
+                        "Record",
+                        1,
+                        t1.into(),
+                        pos1
+                    ));
+                };
+
+                let Term::Record(record2) = t2 else {
+                    return Err(mk_type_error!(
+                        "record/disjoint_merge",
+                        "Record",
+                        2,
+                        t2.into(),
+                        pos2
+                    ));
+                };
+
+                // As for merge, we refuse to combine two records if one of them has a sealed tail
+                if let Some(record::SealedTail { label, .. }) =
+                    record1.sealed_tail.or(record2.sealed_tail)
+                {
+                    return Err(EvalError::IllegalPolymorphicTailAccess {
+                        action: IllegalPolymorphicTailAction::Merge,
+                        evaluated_arg: label.get_evaluated_arg(&self.cache),
+                        label,
+                        call_stack: std::mem::take(&mut self.call_stack),
+                    });
+                }
+
+                // Note that because of record closurization, we assume here that the record data
+                // of each record are already closurized, so we don't really care about
+                // environments. Should that invariant change, we might get into trouble (trouble
+                // meaning undue `UnboundIdentifier` errors).
+                debug_assert!(record1.attrs.closurized && record2.attrs.closurized);
+                record1.fields.extend(record2.fields);
```

Comment:
I don't see where we're checking for disjointness. Should we?

Reply:
It's intentional. The  reasoning was that I want this to be fast and simple especially in various record contract implementation when we *know* that the arguments are disjoints by construction (for example, when re-assembling an unsealed record tail with the rest of the record body).

Checking for disjointedness isn't free, and while it's still fast in Rust land, it kinda diminishes the interest of this function. It's also ok for primops to be unsafe (that is, the caller has to enforce some invariants) IMHO. Lastly, if one ever misuse this function, the worth that can happen is that the field of the right hand side will override the one on the left-hand side (so it's kind of a directed merge). In fact maybe we should brand this primop like this, so that it's not even a prerequisite anymore that the arguments are disjoints?